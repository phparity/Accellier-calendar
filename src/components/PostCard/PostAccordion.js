import React, { useState } from "react";

/**
 * PostAccordion component to render an accordion with a list of items.
 *
 * @param {Object} items - List of items to render in the accordion
 * @return {JSX.Element} The rendered accordion component
 */
function PostAccordion({ items }) {
  // State to keep track of the index of the expanded item
  const [expandedIndex, setExpandedIndex] = useState(null);

  // Function to handle item click and toggle the expanded state
  const handleItemClick = (index) => {
    setExpandedIndex(index === expandedIndex ? null : index);
  };

  // Render the accordion component with the list of items
  return (
    <div className="PostAccordion_otr">
      {items.map((item, index) => (
        <div key={index} className="accordion_item">
          <div
            className={`accordion_header ${index === expandedIndex ? "expanded" : ""}`}
            onClick={() => handleItemClick(index)}
          >
            {item.title}
          </div>
          {index === expandedIndex && (
            <div className="accordion_content">{item.content}</div>
          )}
        </div>
      ))}
    </div>
  );
}

export default PostAccordion;
