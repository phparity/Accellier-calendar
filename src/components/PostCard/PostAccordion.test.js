import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import PostAccordion from "./PostAccordion";

const items = [
  { title: "Title 1", content: "Content 1" },
  { title: "Title 2", content: "Content 2" },
  { title: "Title 3", content: "Content 3" },
];

test("PostAccordion renders properly and expands/collapses items correctly", () => {
  const { getByText, queryByText } = render(<PostAccordion items={items} />);

  items.forEach((item) => {
    expect(queryByText(item.title)).toBeInTheDocument();
    expect(queryByText(item.content)).not.toBeInTheDocument();
  });

  items.forEach((item, index) => {
    const header = getByText(item.title);
    fireEvent.click(header);

    expect(queryByText(item.content)).toBeInTheDocument();

    items
      .filter((_, i) => i !== index)
      .forEach((otherItem) => {
        expect(queryByText(otherItem.content)).not.toBeInTheDocument();
      });

    fireEvent.click(header);
    expect(queryByText(item.content)).not.toBeInTheDocument();
  });
});

describe("Post Accordion Component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<PostAccordion items={items} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
