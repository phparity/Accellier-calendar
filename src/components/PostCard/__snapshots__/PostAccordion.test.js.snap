// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Accordion Component renders correctly 1`] = `
<div
  className="PostAccordion_otr"
>
  <div
    className="accordion_item"
  >
    <div
      className="accordion_header "
      onClick={[Function]}
    >
      Title 1
    </div>
  </div>
  <div
    className="accordion_item"
  >
    <div
      className="accordion_header "
      onClick={[Function]}
    >
      Title 2
    </div>
  </div>
  <div
    className="accordion_item"
  >
    <div
      className="accordion_header "
      onClick={[Function]}
    >
      Title 3
    </div>
  </div>
</div>
`;
